go/vt/grpcclient/compression.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpcclient
+
+import (
+	"fmt"
+	"slices"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/encoding"
+)
+
+type compressionValue string
+
+func (c *compressionValue) String() string {
+	return string(*c)
+}
+
+func (c *compressionValue) Set(d string) error {
+	if slices.Contains(compressionTypes, d) {
+		*c = compressionValue(d)
+	} else {
+		return fmt.Errorf("unsupported compression type: %s", d)
+	}
+	return nil
+}
+
+func (c *compressionValue) Type() string {
```

Comment:
nit: these should be commented. i'm also debating whether this belongs better in `package flagutil` since it's a flag type implementation